src/transformers/tokenization_openai.py

```diff
@@ -213,3 +222,91 @@ def save_vocabulary(self, save_directory):
                 index += 1
 
         return vocab_file, merge_file
+
+
+class _OpenAIGPTCharBPETokenizer(BaseTokenizer):
```

Comment:
Why do we have to have this class here?

Don't we have an implementation of char-level BPE in `tokenizers` now?
Here: https://github.com/huggingface/tokenizers/blob/master/bindings/python/tokenizers/implementations/char_level_bpe.py#L9

Reply:
We do need a special OpenaiGPT implementation because it slightly differs from the char-level BPE we have in tokenizers:

- Normalizer is the same as Bert (BertNormalizer)
- PreTokenizer is not Whitespace, it's the same as Bert (BertPreTokenizer)

Reply:
If we put TransformerXL into tokenizers.implementations, may be this one can make its way to tokenizers too. cc @n1t0

Reply:
Honestly, I'm not too sure about this. I think `tokenizers` should stay a library with some generic implementations, with an easy way for everybody to build it's own custom tokenizer when needed. So I'd like to avoid introducing specific implementations for each new model/tokenizer. Otherwise, the next thing we'll discuss is whether we should have default vocabularies downloaded automatically with each specific implementation, and then we'll have as many implementations as models there are in `transformers`... I think it makes more sense to have specific customization details in `transformers`, next to the model that actually uses the custom tokenizer.

Reply:
> Otherwise, the next thing we'll discuss is whether we should have default vocabularies downloaded automatically with each specific implementation

You mean have all the things that made the success of `Transformers`? 😜

Jking Well ok for me to keep these in `Transformers` then.